src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp

```diff
@@ -111,37 +110,39 @@ ErrorMessageOr<uint64_t> FindFunctionAddressWithFallback(
 
   // Write the name of the .so into memory at code_memory with offset of kCodeScratchPadSize.
   std::vector<uint8_t> path_as_vector(path_length);
-  memcpy(path_as_vector.data(), path.c_str(), path_length);
+  std::memcpy(path_as_vector.data(), path.c_str(), path_length);
   const uint64_t so_path_address = code_memory->GetAddress() + kCodeScratchPadSize;
-  auto write_memory_result = WriteTraceesMemory(pid, so_path_address, path_as_vector);
-  if (write_memory_result.has_error()) {
-    return write_memory_result.error();
-  }
+  OUTCOME_TRY(WriteTraceesMemory(pid, so_path_address, path_as_vector));
 
-  // We want to do the following in the tracee:
-  // return_value = dlopen(path, flag);
-  // The calling convention is to put the parameters in registers rdi and rsi.
-  // So the address of the file path goes to rdi. The flag argument goes into rsi. Then we load the
-  // address of dlopen into rax and do the call. Assembly in Intel syntax (destination first),
-  // machine code on the right:
+  constexpr Lmid_t kLmid = LM_ID_NEWLM;
 
-  // movabsq rdi, so_path_address     48 bf so_path_address
-  // movl esi, flag                   be flag
+  // We want to do the following in the tracee:
+  // return_value = dlmopen(lmid, path, flag)
+  // The calling convention is to put the parameters in registers rdi, rsi, and rdx.
+  // So the lmid goes to rdi, the address of the file path goes to rsi, and the flag argument
+  // goes into rcx. Then we load
```

Comment:
nit: formatting looks broken in this line.

Reply:
Done.